services/notification/notifiers/comment/__init__.py

```diff
@@ -351,18 +352,36 @@ def is_enabled(self) -> bool:
             self.notifier_yaml_settings, dict
         )
 
+    async def is_first_pull(self, comparison: Comparison) -> bool:
+        pull = comparison.pull
+        first_pull = (
```

Comment:
Should we think of caching this value, or you think that we would only ever access `is_first_pull()` roughly once / few times per `Pull` instance?

Reply:
Would be nice to cache somehow.  Can't use `@cached_property` since it needs to cache per comparison - should I maybe just move this method to be defined on `Comparison` instead?

Reply:
I think it makes sense to move to `Comparison`.
I would argue it would make even more sense to move to `Pull` directly, in fact...

Reply:
Added an `is_first_pull` cached property to `Pull` here instead